### DISCUSSION TITLE
How can Call procedure inside a procedure???

### DISCUSSION THREAD

**Discussion by IRediTOTO** [trpc/trpc#3172] [Q&A]
Hello, I have 2 procedure like this
```
export const isSetPassword = procedure.query(()=>{return true})

export const updatePassword= procedure
  .mutation(async ({ ctx, input: { dealId } }) => {
   // How can I call isSetPassword  here?

  });

```

**Comment by jhb-dev** [comment]
I am not sure if it's the best or recommend approach but I am using [Server Side Calls](https://trpc.io/docs/server-side-calls) to achieve this.

**Reply by IRediTOTO** [reply]
No we can't do that. That make loop

**Comment by KATT** [comment]
I never call procedures from other procedures... that's not what they're designed for, they're designed to be called from the outside. 

Make `isSetPassword` a normal function instead and call it

**Reply by IRediTOTO** [reply]
Yes this look like best for now
